### PR TITLE
Refactoring, testing, and build process

### LINKED ISSUE #1: Implement a build process
We at least need to provide a minified version for production.

### PR DESCRIPTION
**Pull request by Nathan-Schwartz**
### Fixes:
 * Competing queries will no longer unexpectedly overwrite each other.
 * Order of queries no longer need to be ascending by pixels.
 * Use uglified version as main file inside package.json
 * Resolves #1 - New build process

### Changes
 * Return all matching queries, not just one.
 * Add respondable.next
 * Remove 'default' value.

### Enhancements
 * Test coverage is over 90%
 * More detailed documentation and comments
 * add .npmignore to get rid of unnecessary files in npm package

### CODE REVIEW

**Comment by brad-decker** [comment]
@Aweary Nudge.

File: src/respondable.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    throw new Error(`Respondable requires an object as its first argument.`);
+  }
+
+  if (!onChangeCb || typeof onChangeCb !== 'function') {
+    throw new Error(`Respondable requires a callback function as its second argument`);
+  }
+
+  if (matchMediaPolyFill !== undefined && typeof matchMediaPolyFill !== 'function') {
+    throw new Error(`Respondable has an optional third parameter for matchMediaPolyFill of type function.`);
+  }
```
Comment (Aweary) [inline src/respondable.js]: Generally, polyfills should be loaded first and be invisible to libraries. It's up to the user to load the appropriate polyfill before they load respondable. You should probably just throw or log a warning and return early if `window.matchMedia` is not defined.

File: src/respondable.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  findMatches(instance);
+
+  state.instances[String(state.nextInstanceID)] = instance;
+  state.nextInstanceID += 1;
+
+  console.warn(`This method will behave differently in the next release. The new implementation is available as 'respondable.next'. Note that 'respondable.next' will not be deprecated in the next version.`);
+  return state.nextInstanceID - 1;
+}
+
+respondable.next = respondableNext;
```
Comment (Aweary) [inline src/respondable.js]: I'd say get rid of the old version and just release as a major version (`1.0.0`) which indicates breaking changes.

File: src/respondable.js (hunk 1 above)
Comment (Nathan-Schwartz) [inline src/respondable.js]: Thanks for the feedback, I completely agree. Fixed in latest push.

File: src/respondable.js (hunk 2 above)
Comment (Nathan-Schwartz) [inline src/respondable.js]: I like it 👍

**Pull request merged**